refinery/core/models.py

```diff
@@ -552,31 +552,17 @@ def delete(self, **kwargs):
         pre_isa_archive associated with the DataSet if one exists.
         """
 
-        try:
-            self.get_isa_archive().delete()
-        except AttributeError as e:
-            logger.debug("DataSet has no isa_archive to delete: %s", e)
-
-        try:
-            self.get_pre_isa_archive().delete()
-        except AttributeError as e:
-            logger.debug("DataSet has no pre_isa_archive to delete: %s", e)
-
-        related_investigation_links = self.get_investigation_links()
-
-        for investigation_link in related_investigation_links:
-            node_collection = investigation_link.get_node_collection()
-            try:
-                node_collection.delete()
-            except Exception as e:
-                logger.error("Couldn't delete NodeCollection: %s", e)
-
```

Comment:
DataSet.delete() should be replaced by pre_ and/or post_delete signal handlers to ensure this functionality is executed on bulk operations
https://docs.djangoproject.com/en/1.8/topics/db/models/#overriding-model-methods